cmake Linking lv_blend_helium.S.o and lv_blend_neon.S.o resulted in errors during compilation.

### LVGL version

v9.2

### What happened?

[ 46%] Building CXX object ../lvgl-9.2.0/out/CMakeFiles/lvgl_thorvg.dir/src/libs/thorvg/tvgLottieExpressions.cpp.o
ld.lld: error: lib/liblvgl.a(lv_blend_helium.S.o) is incompatible with armelf_linux_eabi
ld.lld: error: lib/liblvgl.a(lv_blend_neon.S.o) is incompatible with armelf_linux_eabi
clang-15: error: linker command failed with exit code 1 (use -v to see invocation)

Cross-compiling resulted in errors during linking. Upon checking with the file command, only these two .o files were found to be of x86 architecture, while all the other files are ARM. What’s going on?

### How to reproduce?

_No response_

@kisvegabor I can reproduce the error on Windows x64 (AMD) using Visual Studio 2022:

```
C:\<...>\lvgl\src\widgets\spinbox\lv_spinbox.c(427,56): warning C4267: 'initializing': conversion from 'size_t' to 'const int', possible loss of data 
    lv_spinner.c
    lv_switch.c
    lv_table.c
    lv_tabview.c
    lv_textarea.c
    lv_tileview.c
    lv_win.c
    Generating Code...
    Auto build dll exports
    Couldn't open file 'C:/<...>/lvgl/build/windows-base/lvgl.dir/Debug/lv_blend_helium.obj' with CreateFile()

Build All failed.
```

The repro is to clone the lvgl repo, open the directory using Visual Studio 2022 (Version 17.8.3), and then doing Build->Build All.  Searching I see there is a lv_blend_helium.S with arm assembly in it but there is no lv_blend_helium.c (i.e. the object file appears to for arm only).  So this does indeed look like a cmake issue in that it shouldn't be trying to link in an arm only object file when building for x64.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,10 +3,11 @@ cmake_minimum_required(VERSION 3.12.4)
 set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 
 if(NOT ESP_PLATFORM)
+  project(lvgl LANGUAGES C HOMEPAGE_URL https://github.com/lvgl/lvgl)
   if(NOT (CMAKE_C_COMPILER_ID STREQUAL "MSVC"))
-    project(lvgl LANGUAGES C CXX ASM HOMEPAGE_URL https://github.com/lvgl/lvgl)
+    enable_language(CXX ASM)
   else()
-    project(lvgl LANGUAGES C CXX HOMEPAGE_URL https://github.com/lvgl/lvgl)
+    enable_language(CXX)
   endif()
 endif()
 

